frontend/main/components/ProfilePage/ProfilePage.js

```diff
@@ -0,0 +1,206 @@
+import { BaseComponent } from "../../app/BaseComponent.js";
+import { userinfo } from "./userInfo.js";
+import { ProfileService } from "../../services/ProfileService.js";
+import { EventHub, hub } from "../../eventhub/EventHub.js";
+
+export class ProfilePage extends BaseComponent {
+    constructor() {
+        super();
+        this.container.classList.add('profile-page');
+
+        this.currentUser = userinfo[0]; //only shows the first user now, but will change later
+
+        this.profileContainer = document.createElement("div");
+        this.profileContainer.classList.add("profile-container");
+
+        this.privateInfoVisible = false; 
+        this.loadCSS("ProfilePage");
+        this.render();
+    }
+
+    async getProfileList() {
+        const profileService = new ProfileService();
+        const list = await profileService.retrieveProfileById();
+    }
+
+    render() {
+        this.profileContainer.innerHTML = ""; 
+
+        const publicInfo = this.renderPublicInfo();
+        this.profileContainer.appendChild(publicInfo);
+
+        const privateSections = this.renderPrivateSections();
+        this.profileContainer.appendChild(privateSections);
+
+        const orderHistory = this.renderOrderHistory();
+        this.profileContainer.appendChild(orderHistory);
+
+        this.container.appendChild(this.profileContainer);
+
+        return this.container;
+    }
+
+    renderPublicInfo() {
+        const publicInfo = document.createElement("div");
+        publicInfo.classList.add("public-info");
+
+        //avatar
+        const avatar = document.createElement("img");
+        avatar.src = this.currentUser.avatar;
+        avatar.alt = `${this.currentUser.name}'s Avatar`;
+        avatar.classList.add("profile-avatar");
+        publicInfo.appendChild(avatar);
+
+        //public info: name, phone number, email
+        const userDetails = document.createElement("div");
+        userDetails.classList.add("user-details");
+
+        const name = document.createElement("h1");
+        name.innerText = this.currentUser.name;
+        name.classList.add("profile-name");
+        //edit button for name
+        const editIcon = document.createElement("span");
+        editIcon.classList.add("edit-icon");
+        editIcon.innerText = "✏️";
+
+        editIcon.addEventListener("click", () => {
+        const input = document.createElement("input");
```

Comment:
Can you add styling for this input? currently it appears on top of other text boxes and is difficult to click on.